h2/src/main/org/h2/mvstore/Page.java

```diff
@@ -619,11 +619,18 @@ private void read(ByteBuffer buff) {
             }
             int lenAdd = DataUtils.readVarInt(buff);
             int compLen = buff.remaining();
-            byte[] comp = Utils.newBytes(compLen);
-            buff.get(comp);
+            byte[] comp;
+            int pos = 0;
+            if (buff.hasArray()) {
+                comp = buff.array();
+                pos = buff.position();
```

Comment:
I didn't check it, but I guess it should be `pos = buff.arrayOffset() + buff.position()`. Maybe offset is always 0 with our current code, but who knows how it can be changed in the future.